IOS10-HW14-OZMANYAN AMO/Sourses/ViewController.swift

```diff
@@ -0,0 +1,361 @@
+        collectionView.register(TableCell.self, forCellWithReuseIdentifier: TableCell.identifier)
+        collectionView.register(TableCellHeader.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: TableCellHeader.identifier)
+        collectionView.backgroundColor = .white
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        return collectionView
+    }()
+    
+    // MARK: - Lifecycle
+   
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "МЕДИАТЕКА"
+        setupView()
+        setupHierarchy()
+        setupLayout()
+    }
+       
+    // MARK: - Setup
+    
+    private func setupView() {
+        view.backgroundColor = .white
+    }
+    
+    private func setupHierarchy() {
+        view.addSubview(collection)
+        }
+    
+    private func setupLayout() {
+        NSLayoutConstraint.activate([
+            collection.topAnchor.constraint(equalTo: view.topAnchor),
+            collection.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            collection.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            collection.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+            }
+   
+}
+
+extension ViewController: UICollectionViewDataSource, UICollectionViewDelegate {
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        model.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return model[0].count
+        case 1:
+            return model[1].count
+        case 2:
+            return model[2].count
+        case 3:
+            return model[3].count
+        default:
+            return 0
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let data = model[indexPath.section][indexPath.item]
+        
+        switch indexPath.section {
+        case 0:
+            let item = collectionView.dequeueReusableCell(withReuseIdentifier: AlbumCell.identifier, for: indexPath) as? AlbumCell
+            item?.photoImageView.image = UIImage(named: data.image)
+            item?.namePhotoLabel.text = data.name
+            item?.numberPhotosLabel.text = data.description
+            return item ?? UICollectionViewCell()
+        case 1:
+            let item = collectionView.dequeueReusableCell(withReuseIdentifier: AlbumCell.identifier, for: indexPath) as? AlbumCell
+            item?.photoImageView.image = UIImage(named: data.image)
+            item?.namePhotoLabel.text = data.name
+            item?.numberPhotosLabel.text = data.description
+            return item ?? UICollectionViewCell()
```

Comment:
Тут тоже избавляемся от числовой привязи секций, а ячейки кастим через guard тк это безопасней 
```suggestion
       let sectionType = Sections(rawValue: indexPath.section)

        switch sectionType{
        case .first:
            guard let item = collectionView.dequeueReusableCell(
                withReuseIdentifier: AlbumCell.identifier,
                for: indexPath) as? AlbumCell else {
                
                return UICollectionViewCell()
            }
            item.photoImageView.image = UIImage(named: data.image)
            item.namePhotoLabel.text = data.name
            item.numberPhotosLabel.text = data.description
            return item
        case .second:
            guard let item = collectionView.dequeueReusableCell(
                withReuseIdentifier: AlbumCell.identifier,
                for: indexPath) as? AlbumCell else {

                return UICollectionViewCell()
            }

            item.photoImageView.image = UIImage(named: data.image)
            item.namePhotoLabel.text = data.name
            item.numberPhotosLabel.text = data.description
            return item
```